File: 0274-h-index/0274-h-index.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        papers = [0] * (len(citations)+1)

        for citation in citations:
            if citation >= len(citations):
                papers[-1] += 1
            else:
                papers[citation] += 1
        
        
        h = len(papers)-1
        paper = papers[-1]

        while h > paper:
            h -= 1
            paper += papers[h]
        
        return h

File: 0274-h-index/0274-h-index.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n = static_cast<int>(citations.size());
        std::vector<int> buckets(n + 1, 0);

        // Count papers by capped citation count
        for (int c : citations) {
            if (c >= n) buckets[n]++;   // overflow bucket
            else        buckets[c]++;
        }

        // Accumulate from high to low to find largest h with count >= h
        int count = 0;
        for (int h = n; h >= 0; --h) {
            count += buckets[h];
            if (count >= h) return h;
        }
        return 0; // fallback, though loop will return before this
    }
};